keyring.go

```diff
@@ -196,12 +196,25 @@ func (k KeyRing) VerifyJSONs(ctx context.Context, requests []VerifyJSONRequest)
 
 		keysFetched, err := fetcher.FetchKeys(ctx, keyRequests)
 		if err != nil {
-			return nil, err
+			fetcherLogger.WithError(err).WithField("fetcher", fetcher.FetcherName()).
+				Warn("Failed to request keys from fetcher")
+			continue
 		}
 
 		fetcherLogger.WithField("num_keys_fetched", len(keysFetched)).
 			Info("Got keys from fetcher")
 
+		// For the keys that we successfully fetched, don't try again
+		for resp := range keysFetched {
+			delete(keyRequests, resp)
+		}
+
+		// If we still have outstanding requests that weren't satisfied
+		// then try them with the next key fetcher
+		if len(keyRequests) > 0 {
+			continue
```

Comment:
If we do have other outstanding requests, we drop the keys fetched `keysFetched` to the floor by the looks of it? Don't we need to accumulate them somewhere?

Reply:
Good spot, yes. Fixed.